invokeai/app/invocations/facetools.py

```diff
@@ -0,0 +1,654 @@
+import math
+
+import cv2
+import mediapipe as mp
+import numpy as np
+from PIL import Image, ImageDraw, ImageFilter, ImageOps
+
+from invokeai.app.invocations.baseinvocation import (
+    BaseInvocation,
+    BaseInvocationOutput,
+    InputField,
+    InvocationContext,
+    OutputField,
+    invocation,
+    invocation_output,
+)
+from invokeai.app.invocations.primitives import ImageField, ImageOutput
+from invokeai.app.models.image import ImageCategory, ResourceOrigin
+
+
+@invocation_output("face_mask_output")
+class FaceMaskOutput(BaseInvocationOutput):
+    """Base class for FaceMask output"""
+
+    image: ImageField = OutputField(description="The output image")
+    width: int = OutputField(description="The width of the image in pixels")
+    height: int = OutputField(description="The height of the image in pixels")
+    mask: ImageField = OutputField(description="The output mask")
+
+
+@invocation_output("face_off_output")
+class FaceOffOutput(BaseInvocationOutput):
+    """Base class for FaceOff Output"""
+
+    bounded_image: ImageField = OutputField(default=None, description="Original image bound, cropped, and resized")
+    width: int = OutputField(description="The width of the bounded image in pixels")
+    height: int = OutputField(description="The height of the bounded image in pixels")
+    mask: ImageField = OutputField(default=None, description="The output mask")
+    x: int = OutputField(description="The x coordinate of the bounding box's left side")
+    y: int = OutputField(description="The y coordinate of the bounding box's top side")
```

Comment:
Same comment re: inheriting. Though here, you have changed it slightly with `bounded_image`, so I suppose inheriting doesn't do much for you.